cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js

```diff
@@ -123,21 +204,39 @@ export default class DataPrepTable extends Component {
         <table className="table table-bordered table-striped">
           <thead className="thead-inverse">
             {
-              headers.map((head) => {
+              headers.map((head, index) => {
                 return (
                   <th
-                    id={`column-${head}`}
-                    key={head}
+                    id={`column-${head.name}`}
+                    key={head.name}
                   >
-                    <span className="header-text">
-                      {head}
-                    </span>
-
-                    <span className="float-xs-right directives-dropdown-button">
-                      <ColumnActionsDropdown
-                        column={head}
-                      />
-                    </span>
+                    <div
+                      className="clearfix column-wrapper"
+                    >
+                      {
+                        !head.edit ?
+                          <span
+                            className="header-text float-xs-left"
+                            onClick={this.switchToEditColumnName.bind(this, head)}
+                          >
+                            {head.name}
+                          </span>
+                        :
+                          <input
```

Comment:
Might need to actually separate this out from the table component, so that it doesn't lag out while typing, since it's is diffing the entire table every time we type.